api/filter_dimension_options.go

```diff
@@ -149,46 +161,84 @@ func (api *FilterAPI) getFilterBlueprintDimensionOptionHandler(w http.ResponseWr
 		return
 	}
 
+	bytes, err := json.Marshal(dimensionOption)
+	if err != nil {
+		log.ErrorC("failed to marshal filter blueprint dimension option into bytes", err, logData)
+		if auditErr := api.auditor.Record(r.Context(), getOptionAction, actionUnsuccessful, auditParams); auditErr != nil {
+			handleAuditingFailure(r.Context(), getOptionAction, actionUnsuccessful, w, auditErr, logData)
+			return
+		}
+		http.Error(w, internalError, http.StatusInternalServerError)
+		return
+	}
+
 	if auditErr := api.auditor.Record(r.Context(), getOptionAction, actionSuccessful, auditParams); auditErr != nil {
 		handleAuditingFailure(r.Context(), getOptionAction, actionSuccessful, w, auditErr, logData)
 		return
 	}
 
 	setJSONContentType(w)
-	w.WriteHeader(http.StatusNoContent)
+	w.WriteHeader(http.StatusOK)
+	_, err = w.Write(bytes)
+	if err != nil {
+		log.ErrorC("failed to write bytes for http response", err, logData)
```

Comment:
Wherever `log.ErrorC(...)` exists replace with `log.ErrorCtx(...)`

For example here this can be like so
```
log.ErrorCtx(r.Context(), errors.Wrap(err, "failed to write bytes for http response"), logData)
```

Reply:
Using this log error type extracts the request id stored in context and outputs this in the logs, allowing us to trace requests through codebase.